Code/Framework/AzFramework/Platform/Linux/AzFramework/Components/NativeUISystemComponent_Linux.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/Application/Application.h>
+#include <AzFramework/Components/NativeUISystemComponent.h>
+
+#include <sys/resource.h>
+
+#if PAL_TRAIT_LINUX_WINDOW_MANAGER_XCB
+#include <AzFramework/XcbApplication.h>
+#include <AzFramework/XcbInputDeviceKeyboard.h>
+#include <AzFramework/XcbInputDeviceMouse.h>
+#include <AzFramework/XcbNativeWindow.h>
+#endif
+
+constexpr rlim_t g_minimumOpenFileHandles = 65536L;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+namespace AzFramework
+{
+    struct LinuxApplicationImplFactory 
+        : public Application::ImplementationFactory
+    {
+        Application::Implementation* Create() override
```

Comment:
All of the `<device-type>::ImplementationFactory` overrides member functions in this file should return a unique_ptr instead of a raw pointer